geps/gep-1897.md

```diff
@@ -0,0 +1,172 @@
+
+In February 2020, [HTTPRoute: Add Reencrypt #81](https://github.com/kubernetes-sigs/gateway-api/pull/81)
+added the dataplane feature as “reencrypt”, but it went stale and was closed in favor of the work done in the
+next paragraph, which unfortunately didn’t implement the backend TLS termination feature.
+
+In August 2020, it resurfaced with a [comment](https://github.com/kubernetes-sigs/gateway-api/pull/256/files#r472734392)
+on this pull request: [tls: introduce mode and sni to cert matching behavior](https://github.com/kubernetes-sigs/gateway-api/pull/256/files#top).
+The backend TLS termination feature was deferred at that time.  Other TLS discussion was documented in
+[[SIG-NETWORK] TLS config in service-apis](https://docs.google.com/document/d/15fkzMrhN_7tA-i2mHKwZpqcjN1o2Pe9Am9Qt828x1lo/edit#heading=h.wym7wehwll44)
+, a list of TLS features that had been collected in June 2020, itself based on spreadsheet
+[Service API: TLS related issues](https://docs.google.com/spreadsheets/d/18KE61Y6InCmoQHZcbrYYRZS5Cnt7n33s5dTxUlhHgIA/edit#gid=0).
+
+In December 2021, this was discussed as a beta blocker in issue
+[Docs mentions Reencrypt for HTTPRoute and TLSRoute is available #968](https://github.com/kubernetes-sigs/gateway-api/issues/968).
+
+A March 2022 issue documents another request for it: [Provide a way to configure TLS from a Gateway to Backends #1067](https://github.com/kubernetes-sigs/gateway-api/issues/1067)
+
+A June 2022 issue documents a documentation issue related to it:
+[Unclear how to specify upstream (webserver) HTTP protocol #1244](https://github.com/kubernetes-sigs/gateway-api/discussions/1244)
+
+A July 2022 discussion [Specify Re-encrypt TLS Termination (i.e., Upstream TLS) #1285](https://github.com/kubernetes-sigs/gateway-api/discussions/1285)
+collected most of the historical context preceding the backend TLS termination feature, with the intention of
+collecting evidence that this feature is still unresolved.  This was followed by
+[GEP: Describe Backend Properties #1282](https://github.com/kubernetes-sigs/gateway-api/issues/1282).
+
+In August 2022, [Add Provisional GEP-1282 document #1333](https://github.com/kubernetes-sigs/gateway-api/pull/1333)
+was created, and in October 2022, a GEP update with proposed implementation
+[GEP-1282 Backend Properties - Update implementation #1430](https://github.com/kubernetes-sigs/gateway-api/pull/1430)
+was followed by intense discussion and closed in favor of a downsize in scope.
+
+In January 2023 we closed GEP-1282 and began a new discussion on enumerating TLS use cases in
+[Gateway API TLS Use Cases](#references), for the purposes of a clear definition and separation of concerns.
+This GEP is the outcome of the TLS use cases #4 and #5 in
+[Gateway API TLS Use Cases](#references) as mentioned in the Goals section above.
+
+## API
+
+Details deferred until we reach consensus on what we want to do, and why we want to do this.
+
+## Prior Art
```

Comment:
So:

* Client decides TLS: Istio, Ambassador, OpenShift
* Server (service) decides TLS: Contour, GKE

Is that a fair representation?

Reply:
I think Ambassador relies on the service configuration, so I would put it with the second bullet.

Reply:
AFAIK Istio is both - with the new auto-mtls clearly in 'server decide' category.

And the service ultimately decides everywhere - if the service has a TLS port, there is nothing a client can do to change the fact that only TLS connections will work.

Reply:
@howardjohn @costin If there's something you want me to add to summarize this better, or you want me to point out the two categorizations, please let me know.